File: IncomeTax_cal.java
package Variables;

import java.util.Scanner;

public class IncomeTax_cal {

    public static void main(String[] args) {
        // Create scanner object to take input
        Scanner input = new Scanner(System.in);

        // Prompt the user to enter their income
        System.out.println("Enter your Income:");
        int income = input.nextInt();

        // Variable to store the calculated tax
        int tax = 0;

        // Income tax slabs
        if (income <= 250000) {
            tax = 0; // No tax for income up to 2.5 lakh
        }
        else if (income >= 250001 && income <= 500000) {
            tax = (int) (income * 0.05); // 5% tax for income between 2.5 lakh to 5 lakh
        }
        else if (income >= 500001 && income <= 1000000) {
            tax = (int) (income * 0.20); // 20% tax for income between 5 lakh to 10 lakh
        }
        else {
            tax = (int) (income * 0.30); // 30% tax for income above 10 lakh
        }

        // Output the calculated tax
        System.out.println("Tax is: " + tax);
    }
}

File: Income_Tax_Cal.py
def calculate_tax(income):
    tax = 0

    # Income tax slabs and rates
    if income <= 250000:
        tax = 0  # No tax for income up to ₹2.5 lakh
    elif income <= 500000:
        tax = (income - 250000) * 0.05  # 5% tax for income between ₹2.5 lakh to ₹5 lakh
    elif income <= 1000000:
        tax = (250000 * 0.05)  # 5% tax for the first ₹2.5 lakh
        tax += (income - 500000) * 0.20  # 20% tax for income between ₹5 lakh to ₹10 lakh
    else:
        tax = (250000 * 0.05)  # 5% tax for the first ₹2.5 lakh
        tax += (500000 * 0.20)  # 20% tax for the next ₹5 lakh
        tax += (income - 1000000) * 0.30  # 30% tax for income above ₹10 lakh

    return tax


def main():
    # Take income input from the user
    income = int(input("Enter your Income: "))
    
    # Calculate the tax based on the income
    tax = calculate_tax(income)

    # Output the calculated tax
    print(f"Tax is: ₹{int(tax)}")


if __name__ == "__main__":
    main()
